AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -668,4 +673,115 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
+                                         appSecret:self.appSecret
+                                         partition:kMSDataStoreUserDocumentsPartition
+                               includeExpiredToken:NO
+                                 completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+                                   if (error) {
+                                     MSLogError([MSDataStore logTag], @"Cannot read from local storage because there is no "
+                                                                      @"account ID cached and failed to retrieve token.");
+                                     return;
+                                   }
+
+                                   // Run the operation in a dispatch queue.
+                                   dispatch_async(self.dispatchQueue, ^{
+                                     NSArray<MSPendingOperation *> *pendingOperations =
+                                         [self.dataOperationProxy.documentStore pendingOperationsWithToken:tokenResponses.tokens[0]];
+                                     for (MSPendingOperation *operation in pendingOperations) {
+                                       MSWriteOptions *writeOptions =
+                                           [[MSWriteOptions alloc] initWithDeviceTimeToLive:operation.getDeviceTimeToLiveFromOperation];
+                                       if ([operation.operation isEqualToString:kMSPendingOperationCreate] ||
+                                           [operation.operation isEqualToString:kMSPendingOperationReplace]) {
+
+                                         // Perform create and update operation
+                                         [MSDataStore
+                                             replaceWithPartition:(NSString *)operation.partition
+                                                       documentId:(NSString *)operation.documentId
+                                                         document:(id<MSSerializableDocument>)operation.document
+                                                     writeOptions:(MSWriteOptions * _Nullable) writeOptions
+                                                completionHandler:^(MSDocumentWrapper *documentWrapper) {
+                                                  if (!documentWrapper.error) {
+                                                    if (writeOptions.deviceTimeToLive > kMSDataStoreTimeToLiveNoCache ||
+                                                        writeOptions.deviceTimeToLive == kMSDataStoreTimeToLiveInfinite) {
+                                                      [self.dataOperationProxy.documentStore upsertWithToken:tokenResponses.tokens[0]
+                                                                                             documentWrapper:documentWrapper
+                                                                                                   operation:nil
```

Comment:
Suggestion: It would be helpful to have a comment explain why we're setting `pending_operation` column to `nil` here.